agnpy/wrappers/gammapy.py

```diff
@@ -0,0 +1,208 @@
+        B = Parameter("B", blob.B)
+        R_b = Parameter("R_b", blob.R_b)
+        parameters.extend([z, d_L, delta_D, B, R_b])
+
+        self.default_parameters = Parameters(parameters)
+        super().__init__()
+
+    def evaluate(self, energy, **kwargs):
+        """evaluate"""
+        nu = energy.to("Hz", equivalencies=u.spectral())
+
+        # all the model parameters will be passed as kwargs, by SpectralModel.evaluate()
+        # sort the ones related to the source out
+        z = kwargs.pop("z")
+        d_L = kwargs.pop("d_L")
+        delta_D = kwargs.pop("delta_D")
+        B = kwargs.pop("B")
+        R_b = kwargs.pop("R_b")
+
+        # expand the remaining kwargs in the spectral parameters
+        args = kwargs.values()
+
+        sed_synch = Synchrotron.evaluate_sed_flux(
+            nu, z, d_L, delta_D, B, R_b, self.blob.n_e, *args
+        )
+        sed_ssc = SynchrotronSelfCompton.evaluate_sed_flux(
+            nu, z, d_L, delta_D, B, R_b, self.blob.n_e, *args
+        )
+
+        sed = sed_synch + sed_ssc
+
+        # to avoid the problem pointed out in
+        # https://github.com/cosimoNigro/agnpy/issues/117
+        # we can do here something like
+        # sed = sed.reshape(energy.shape)
+        # this is done in Gammapy's `NaimaSpectralModel`
+        # https://github.com/gammapy/gammapy/blob/master/gammapy/modeling/models/spectral.py#L2119
+
+        # gammapy requires a differential flux in input
+        return (sed / energy ** 2).to("1 / (cm2 eV s)")
+
+
+class SpectralModelEC(SpectralModel):
+
+    tag = ["SpectralModelEC"]
+
+    def __init__(self, blob, blr, dt, r, ec_blr=True):
+        """Initialise all the parameters from the Blob and the targets instances."""
+
+        self.blob = blob
+        self.blr = blr
+        self.dt = dt
+        self.ec_blr = ec_blr
+
+        # check for the two targets to have the same disk luminosities
+        if not u.isclose(self.blr.L_disk, self.dt.L_disk):
+            raise ValueError(
+                "The BLR and DT provided have the different disk luminosities."
+            )
+
+        parameters = []
+
+        # all the parameters of the EED have to be parameters of the fittable model
+        # any electron distribution has for attributes the spectral parameters and the integrator
+        # we remove the latter
+        pars = vars(self.blob.n_e)
+        pars.pop("integrator")
+        for name in pars.keys():
+            parameter = Parameter(name, pars[name])
+            parameters.append(parameter)
+
+        # emission region parameters
+        z = Parameter("z", blob.z)
+        d_L = Parameter("d_L", blob.d_L)
+        delta_D = Parameter("delta_D", blob.delta_D)
+        B = Parameter("B", blob.B)
+        R_b = Parameter("R_b", blob.R_b)
+        mu_s = Parameter("mu_s", blob.mu_s)
+        # distance of the emission region along the jet
+        r = Parameter("r", r)
+        parameters.extend([z, d_L, delta_D, B, R_b, mu_s, r])
```

Comment:
maybe adding something like
self.emission_region_parameters
self.targets_parameters
could help to simplify the sorting of the parameters later in evaluate
(and could also be convenient for user that want to look only at one group of parameters)

Reply:
I grouped the parameters following your suggestions, I don't know though how to re-use this grouping in the `evaluate` function. I see that the `NaimaSpectralModel` in `Gammapy` does something similar, i.e. uses in the `evaluate` some attribute of the class, but then this `_update_naima_parameters` is needed https://github.com/gammapy/gammapy/blob/master/gammapy/modeling/models/spectral.py#L2107. Let me know if you have some suggestion here.